src/app/src/pages/eval-creator/components/ProviderConfigDialog.tsx

```
@@ -35,13 +75,203 @@ const ProviderConfigDialog: React.FC<ProviderConfigDialogProps> = ({
+              // Field was cleared, remove it
+              delete configToSave[key];
+            } else {
+              configToSave[key] = localConfig[key];
+            }
+          }
+        });
+
+        // Handle deployment_id for Azure providers
+        if (isAzureProvider && 'deployment_id' in localConfig) {
+          if (localConfig.deployment_id === undefined) {
+            delete configToSave.deployment_id;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)